core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -460,7 +463,15 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
-    return ArrayUtil.toLongList(splitOffsets);
+    if (splitOffsetsAsList == null && splitOffsets != null) {
+      synchronized (this) {
```

Comment:
I wonder whether we can drop `synchronized` here. It is unlikely we would access this from multiple threads and even if we do, there would be no correctness issues (at most, we would convert the array multiple times). Not sure whether the extra overhead of acquiring the lock all the time is worth it.

Reply:
That's my feeling too, I'll update.

Reply:
No strong opinion, if we drop, should we add a comment on the method?